ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/AttestationUtil.java

```diff
@@ -98,6 +100,121 @@
+    UnsignedLong current_epoch = BeaconStateUtil.get_current_epoch(state);
+
+    // Get current_epoch_attestations
+    List<PendingAttestation> current_epoch_attestations =
+        AttestationUtil.get_epoch_attestations(state, current_epoch);
+
+    // Get current epoch_boundary_attestations
+    List<PendingAttestation> current_epoch_boundary_attestations =
+        AttestationUtil.get_current_epoch_boundary_attestations(state, current_epoch_attestations);
+
+    // Get current_epoch_boundary_attester_indices
+    List<Integer> current_epoch_boundary_attester_indices =
+        AttestationUtil.get_attester_indices(state, current_epoch_boundary_attestations);
+
+    // Get current_epoch_boundary_attesting_balance
+    UnsignedLong current_epoch_boundary_attesting_balance =
+        AttestationUtil.get_total_attesting_balance(state, current_epoch_boundary_attester_indices);
+
+    return current_epoch_boundary_attesting_balance;
+  }
+
+  /**
+   * Returns the sum of balances for all the attesters that were active at the previous epoch
+   * boundary
+   *
+   * @param state
+   * @return previous_epoch_boundary_attesting_balance
+   */
+  public static UnsignedLong get_previous_epoch_boundary_attesting_balance(BeaconState state)
+      throws Exception {
+
+    // Get previous epoch
+    UnsignedLong previous_epoch = BeaconStateUtil.get_previous_epoch(state);
+
+    // Get previous_epoch_attestations
+    List<PendingAttestation> previous_epoch_attestations =
+        AttestationUtil.get_epoch_attestations(state, previous_epoch);
+
+    // Get previous_epoch_boundary_attestations
+    List<PendingAttestation> previous_epoch_boundary_attestations =
+        AttestationUtil.get_previous_epoch_boundary_attestations(
+            state, previous_epoch_attestations);
+
+    // Get previous_epoch_boundary_attester_indices
+    List<Integer> previous_epoch_boundary_attester_indices =
+        AttestationUtil.get_attester_indices(state, previous_epoch_boundary_attestations);
+
+    // Get previous_epoch_boundary_attesting_balance
+    UnsignedLong previous_epoch_boundary_attesting_balance =
+        AttestationUtil.get_total_attesting_balance(
+            state, previous_epoch_boundary_attester_indices);
+
+    return previous_epoch_boundary_attesting_balance;
+  }
+
+  public static List<Integer> get_previous_epoch_head_attester_indices(BeaconState state) {
+    // todo
+    return new ArrayList<Integer>();
+  }
+
+  public static UnsignedLong get_previous_epoch_head_attesting_balance(BeaconState state) {
+    // todo
+    return UnsignedLong.ZERO;
+  }
+
+  public static List<Integer> get_previous_epoch_attester_indices(BeaconState state) {
+    // todo
+    return new ArrayList<Integer>();
+  }
+
+  public static UnsignedLong inclusion_distance(BeaconState state, Integer index) {
+    // todo
+    return UnsignedLong.ZERO;
+  }
+
+  private static boolean isAttestationCurrentEpoch(BeaconState state, PendingAttestation record) {
```

Comment:
Is this being used anywhere? I had deleted it early because it seemed to be outdated and unnecessary

Reply:
I'll delete it.